### PR TITLE
Added ReX reboot snippet

### PR DESCRIPTION
**Pull request by lzap**
SSIA

We saw behavior when the task is reexecuted again, we are still investigating the root cause of this.

### CODE REVIEW

**Comment by mmoll** [comment]
I saw something similar and I think the cause was that the task was still in the proxy-side tasks DB as "to execute".

File: provisioning_templates/snippet/remote_execution_reboot.erb [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+# Proper reboot of a host. It is not recommended to use "shutdown" command
+# directly as it kills SSH connection and Remote Execution may mark the job
+# as failed which might lead to reexecution in the worst case.
+#
+# Parameters:
+#
+# remote_execution_minutes_to_reboot: number of minutes to reboot (default 1)
+#
+
+<% min = host_param('remote_execution_minutes_to_reboot') || 1 %>
```
Comment (ekohl) [inline provisioning_templates/snippet/remote_execution_reboot.erb:15]: Does it make sense to do this in seconds? A minute is a long time and I can imagine some users might want to drop to 30 seconds.

**Comment by iNecas** [comment]
I don't think this code itself has the relation to the re-execution of the commands, compared to the fact that there were other factors in the reproducing environment (such as periodic restarts for smart_proxy_dynflow_core), that could lead to it. There already is a template for that here https://github.com/theforeman/community-templates/blob/develop/job_templates/power_action_-_ssh_default.erb. Anyway, I would deffer with this until we have confirmed that this is the reason for the observed behavior, as I'm not convinced that this would have any effect on that at the moment.

**Comment by lzap** [comment]
I did not know that ReX carries own templates in its git repo. Closing.

**Comment by ares** [comment]
@lzap just to clarify, templates in REX repo should be in sync with what we have here, it's the same model that Foreman uses. If this snippet is useful, it belongs here.

**Pull request closed without merging**